pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -0,0 +1,241 @@
+/*
+Bloom-compactor
+
+This is a standalone service that is responsible for compacting TSDB indexes into bloomfilters.
+It creates and merges bloomfilters into an aggregated form, called bloom-blocks.
+It maintains a list of references between bloom-blocks and TSDB indexes in files called meta.jsons.
+
+Bloom-compactor regularly runs to check for changes in meta.jsons and runs compaction only upon changes in TSDBs.
+
+bloomCompactor.Compactor
+
+			| // Read/Write path
+		bloomshipper.Store**
+			|
+		bloomshipper.Shipper
+			|
+		bloomshipper.BloomClient
+			|
+		ObjectClient
+			|
+	.....................service boundary
+			|
+		object storage
+*/
+package bloomcompactor
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+	"github.com/owen-d/BoomFilters/boom"
+	"github.com/prometheus/client_golang/prometheus"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/grafana/loki/pkg/storage"
+	"github.com/grafana/loki/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/pkg/storage/config"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/bloomshipper"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/bloomshipper/bloomshipperconfig"
+)
+
+type Compactor struct {
+	services.Service
+
+	cfg                Config
+	logger             log.Logger
+	bloomCompactorRing ring.ReadRing
+	periodConfigs      []config.PeriodConfig
+
+	// temporary workaround until store has implemented read/write shipper interface
+	bloomShipperClient bloomshipper.Client
+	bloomStore         bloomshipper.Store
+}
+
+func New(cfg Config,
+	readRing ring.ReadRing,
+	storageCfg storage.Config,
+	periodConfigs []config.PeriodConfig,
+	logger log.Logger,
+	clientMetrics storage.ClientMetrics,
+	_ prometheus.Registerer) (*Compactor, error) {
+	c := &Compactor{
+		cfg:                cfg,
+		logger:             logger,
+		bloomCompactorRing: readRing,
+		periodConfigs:      periodConfigs,
+	}
+
+	//objectClient, err := storage.NewObjectClient(storageCfg.TSDBShipperConfig.SharedStoreType, storageCfg, clientMetrics)
+	//if err != nil { return nil, err}
+	//
+
+	client, err := bloomshipper.NewBloomClient(periodConfigs, storageCfg, clientMetrics)
+	if err != nil {
+		return nil, err
+	}
+
+	shipper, err := bloomshipper.NewShipper(
+		client,
+		bloomshipperconfig.Config{WorkingDirectory: cfg.WorkingDirectory},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	store, err := bloomshipper.NewBloomStore(*shipper)
```

Comment:
NewBloomStore should take a pointer

```suggestion
	store, err := bloomshipper.NewBloomStore(shipper)
```